### DISCUSSION TITLE
when i drag  links in google.com  ,  the backgroup open page is not be read as the sub-tab in TST

### DISCUSSION THREAD

**Discussion by whysowt326** [piroor/treestyletab#3510] [General]
when i drag one link in google result page
 ,  the background open page is be read as the same level from this google page, instead of a sub tab. Unless you right click the link , and choose"open the link in new tap".
what i want is to keep those new tap which is from the link i drag and drop( in the firefox , not the TST) to be sub tab from the original page.  Imagine this scenario , i search Tylor Swift via google server , and there is multiple link in result page , and then i will drag multiple link and open them in background new tab , i want all this tab be in the sub position of the google result page.

![2024-03-26_16-59-02](https://github.com/piroor/treestyletab/assets/147795230/e458774c-8171-4624-abc1-e3c2766490ee)
![2024-03-26_16-59-52](https://github.com/piroor/treestyletab/assets/147795230/52b95492-7acf-46a1-a6bd-a08518975d22)

**Comment by piroor** [comment]
I couldn't reproduce the problem - all tabs opened from the Google search result page become children of the tab of the Google search result page. Dragging a search result link and dropping it on the sidebar opens the link at the specified position. Do you use any other addon which provides ability to open tabs by drag and drop?

**Reply by whysowt326** [reply]
yes, i use the fire-drag https://github.com/erictsangx/fire-drag , because i use Firefox
do you mean the TST itself support the drap and open links in backgroud? it is new information to me indeed.

**Reply by piroor** [reply]
Thanks. This need to be fixed by the fire-drag: it need to call the API `tabs.create()` with an extra parameter `openerTabId`.
https://github.com/erictsangx/fire-drag/blob/f5d865e148895adc2f21a3098d0f70a39d4ae844/src/extension/background.js#L39

**Reply by piroor** [reply]
I already sent a pull request to do that https://github.com/erictsangx/fire-drag/pull/37 but no response from the author...
Maybe the project was abandoned and need to be forked by someone successor.

**Reply by whysowt326** [reply]
do you have some recommend information about any other drap addons working seamlessly with TST?

**Reply by piroor** [reply]
Sorry I have no such information... I generally search keywords via the context menu and open links by middle click.

**Reply by whysowt326** [reply]
https://github.com/harytfw/GlitterDrag, it is the alternative , alright after 2 hours testing, anyway , thanks for your information to let me know problem maybe from the drag addons .Have a nice day .

**Comment by whysowt326** [comment]
![2024-03-28_12-38-34](https://github.com/piroor/treestyletab/assets/147795230/c7e79ba3-bef2-4e4b-8934-4cfb181b4560)
my setting page for the firedrag

**Comment by whysowt326** [comment]
when click or ctrl+click for the link on google search result page, that is okay to be as the children tab .
but when i use the drag, it only be as the children tab when has similar domain in the address bar , as setted in your TST configuration.
